### PR TITLE
Migrate tests from "model/preference_test.go" to use testify

### PR DESCRIPTION
**Pull request by MayMeow**
#### Summary

Update `preference_test.go` to use `require` package instead of `t.fatal`

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/12638

### CODE REVIEW

File: model/preference_test.go [hunk 1]
```diff
@@ -57,8 +57,9 @@ func TestPreferencePreUpdate(t *testing.T) {
 	preference.PreUpdate()
 
 	var props map[string]string
-	if err := json.NewDecoder(strings.NewReader(preference.Value)).Decode(&props); err != nil {
-		t.Fatal(err)
+	err := json.NewDecoder(strings.NewReader(preference.Value)).Decode(&props)
+	if err != nil {
```
Comment (niklabh) [inline model/preference_test.go]: if check is not needed. require.Nil will take care of that.

Comment (MayMeow) [reply model/preference_test.go]: ok updated it

**Comment by jespino** [comment]
QA not needed, only modifies unit tests.

**Comment by jespino** [comment]
Merged! Thanks @MayMeow! :tada:

**Pull request merged**